recipe/meta.yaml

```
@@ -1,42 +1,57 @@
 {% set name = "bidict" %}
-{% set version = "0.21.2" %}
-{% set compress_type = "tar.gz" %}
-{% set hash_val = "4fa46f7ff96dc244abfc437383d987404ae861df797e2fd5b190e233c302be09" %}
+{% set version = "0.23.1" %}
+{% set commit_id = "636f5a610f0e0b9a1e056fc72b688720c114403b" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  fn: {{ name }}-{{ version }}.{{ compress_type }}
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: {{ hash_val }}
+  - url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+    sha256: 03069d763bc387bbd20e7d49914e75fc4132a41937fa3405417e1a5a2d006d71
+    # tests and tests data files
+  - url: https://github.com/jab/bidict/archive/{{ commit_id }}.tar.gz
+    sha256: 97c6b21db087960ed54752aaaf800cd52fe2f31cc3d48efde727c43e1704bf8d
+    folder: tests
```

What this line means is that you are extracting the GH tarball in a `tests` directory.  That's kind-of OK but the GH tarball is really the whole source distribution _inside which_ is the `tests` directory that actually has the tests.  So, nominally, `tests/tests` contains the tests!

More commonly you might say that this (tarball extraction) want to be done into a `gh_src` directory (as opposed to the pypi tarball being extracted into a `work` directory) and then the actual tests will be in `gh_src/tests` which is slightly less confusing!

```suggestion
    folder: gh_src
```

There is a knock-on effect below in the `tests` section

---

Got you, I delete this field according to comment from Andrii Osipov, but will know this information for the future. Thank you for explanation.